packages/auto-pr-labeler/src/mutations.ts

```diff
@@ -0,0 +1,148 @@
+							nodes {
+								name
+							}
+						}
+					}
+				}
+			}
+	`;
+
+const removeLabelsMutation = `
+			mutation {
+				removeLabelsFromLabelable(
+					input: {
+						labelIds: labelIds,
+						labelableId: labelableId
+					}
+				) {
+					labelable {
+						labels(first: 10) {
+							nodes {
+								name
+							}
+						}
+					}
+				}
+			}
+	`;
+
+const assignLabelsAfterClose = async (
+	labelableId: string,
+	token: string
+): Promise<GraphQlQueryResponse<LabelsQueryResponse>> => {
+	const labelIds = [labels.statusInvalid];
+	return await graphqlWithAuth(token)(addLabelsMutation, { labelIds, labelableId });
+};
+
+const assignLabelsAfterMerge = async (
+	labelableId: string,
+	token: string
+): Promise<GraphQlQueryResponse<LabelsQueryResponse>> => {
+	const labelIds = [labels.statusCompleted];
+	return await graphqlWithAuth(token)(addLabelsMutation, { labelIds, labelableId });
+};
+
+const assignLabelsAfterCreated = async (
+	labelableId: string,
+	token: string
+): Promise<GraphQlQueryResponse<LabelsQueryResponse>> => {
+	const labelIds = [labels.statusNew];
+	return await graphqlWithAuth(token)(addLabelsMutation, { labelIds, labelableId });
+};
+
+const assignLabelsAfterReviewApproved = async (
+	labelableId: string,
+	token: string
+): Promise<GraphQlQueryResponse<LabelsQueryResponse>> => {
+	const labelIds = [labels.statusApproved];
+	return await graphqlWithAuth(token)(addLabelsMutation, { labelIds, labelableId });
+};
+
+const assignLabelsAfterReviewChangesRequested = async (
+	labelableId: string,
+	token: string
+): Promise<GraphQlQueryResponse<LabelsQueryResponse>> => {
+	const labelIds = [labels.statusPleaseFix];
+	return await graphqlWithAuth(token)(addLabelsMutation, { labelIds, labelableId });
+};
+
+const assignLabelsAfterReviewRequested = async (
+	labelableId: string,
+	token: string
+): Promise<GraphQlQueryResponse<LabelsQueryResponse>> => {
+	const labelIds = [labels.statusCodeReview];
+	return await graphqlWithAuth(token)(addLabelsMutation, { labelIds, labelableId });
+};
+
+const removeAllStatusLabels = async (
+	labelableId: string,
+	token: string
+): Promise<GraphQlQueryResponse<LabelsQueryResponse>> => {
+	const labelIds = [
+		labels.statusNew,
+		labels.statusPlanning,
+		labels.statusNeedsFeedback,
+		labels.statusInProgress,
+		labels.statusCodeReview,
+		labels.statusPleaseFix,
+		labels.statusApproved,
+		labels.statusNeedsTesting,
+		labels.statusCompleted,
+		labels.statusBlocked,
+		labels.statusDuplicate,
+		labels.statusInvalid,
+	];
+	return await graphqlWithAuth(token)(removeLabelsMutation, { labelIds, labelableId });
+};
+
+export const assignStatusLabels = async (pullRequest: PullRequest, token: string): Promise<void> => {
+	// eslint-disable-next-line no-console
+	console.log('%c pullRequest', 'color: HotPink;', pullRequest);
+	await removeAllStatusLabels(pullRequest.id, token);
+	switch (pullRequest.state) {
+		case 'OPEN':
+			switch (pullRequest.reviewDecision) {
```

Comment:
Sorry I'm confused on this part, the pullRequest.reviewDecision data, if how the automation works to fetch the status of the PR.

Reply:
added a comment regarding that

Reply:
but basically if you go here:
https://docs.github.com/en/graphql/overview/explorer

and sign in with your github account

then enter the following query into the top code box:

```js
query ($pr: Int!, $org: String!, $repo: String!) {
	repository(name: $repo, owner: $org) {
		pullRequest(number: $pr) {
			id
			labels(first: 10) {
				nodes {
					name
				}
			}
			number
			reviewDecision
			state
			closingIssuesReferences(first: 10) {
				nodes {
					id
					number
				}
			}
		}
	}
}
```

then enter the following for the query variables:

```js
{"org": "eventespresso", "repo":"barista", "pr":1098}
```

then run the query, you will see those parameters in the return data

Reply:
oh the other thing that may be confusing is because this is a custom action...
in order for it to work it will require ANOTHER action to be added in the appropriate repo, and THAT action will supply the PR number, etc when the action is triggered